app/plugin_api_tests/test_db_driver/main.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main_test
+
+import (
+	"database/sql"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/v5/app/plugin_api_tests"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/mattermost/mattermost-server/v5/shared/driver"
+	"github.com/mattermost/mattermost-server/v5/store/sqlstore"
+	"github.com/mattermost/mattermost-server/v5/store/storetest"
+)
+
+type MyPlugin struct {
+	plugin.MattermostPlugin
+	config plugin_api_tests.BasicConfig
+	t      *testing.T
+}
+
+func (p *MyPlugin) OnConfigurationChange() error {
+	if err := p.API.LoadPluginConfiguration(&p.config); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *MyPlugin) MessageWillBePosted(_ *plugin.Context, _ *model.Post) (*model.Post, string) {
+	store := sqlstore.New(p.API.GetUnsanitizedConfig().SqlSettings, nil)
+	store.GetMaster().Db.Close()
```

Comment:
Hmm, I couldn't get why this is needed?

Reply:
We just needed the `sqlstore.SqlStore` struct, but with the DB object replaced with our own. The `sqlstore.New` function doesn't allow us to pass our own DB. So we need to create the DB object, close it, and replace it with our own.

Reply:
Oh.. I see thanks 👍